app/controllers/puppetca_controller.rb

```diff
@@ -1,15 +1,13 @@
 class PuppetcaController < ApplicationController
+  rescue_from ProxyAPI::ProxyException do |exception|
```

Comment:
Why only in this controller?  I somewhat see your point about the 500 page (noting that the behaviour is different between prod and dev too, prod is prettier), but shouldn't this be in app controller and handled application-wide?